core/blockchain/impl/block_storage_impl.cpp

```diff
@@ -216,28 +220,80 @@ namespace kagome::blockchain {
     return outcome::success();
   }
 
-  outcome::result<void> KeyValueBlockStorage::removeBlock(
-      const primitives::BlockHash &hash,
-      const primitives::BlockNumber &number) {
-    auto block_lookup_key = numberAndHashToLookupKey(number, hash);
-    auto header_lookup_key = prependPrefix(block_lookup_key, Prefix::HEADER);
-    if (auto rm_res = storage_->remove(header_lookup_key); !rm_res) {
-      logger_->error("could not remove header from the storage: {}",
-                     rm_res.error().message());
-      return rm_res;
+  outcome::result<void> BlockStorageImpl::removeBlock(
+      const primitives::BlockInfo &block) {
+    auto block_lookup_key = numberAndHashToLookupKey(block.number, block.hash);
+
+    SL_TRACE(logger_, "Removing block {}...", block);
+
+    auto hash_to_idx_key =
+        prependPrefix(Buffer{block.hash}, Prefix::ID_TO_LOOKUP_KEY);
+    if (auto res = storage_->remove(hash_to_idx_key); res.has_error()) {
+      logger_->error("could not remove hash-to-idx from the storage: {}",
+                     res.error().message());
+      return res;
+    }
+
+    auto num_to_idx_key =
+        prependPrefix(numberToIndexKey(block.number), Prefix::ID_TO_LOOKUP_KEY);
+    OUTCOME_TRY(num_to_idx_val_opt, storage_->tryGet(num_to_idx_key));
+    if (num_to_idx_val_opt == block_lookup_key) {
+      if (auto res = storage_->remove(num_to_idx_key); res.has_error()) {
+        SL_ERROR(logger_,
+                 "could not remove num-to-idx from the storage: {}",
+                 block,
+                 res.error().message());
+        return res;
+      }
+    }
+
+    // TODO(xDimon): needed to clean up trie storage if block deleted
+    // auto trie_node_lookup_key =
```

Comment:
Please get rid of commented lines - either uncomment or remove them.
Otherwise, it is unclear when to proceed with them

Reply:
Yep. Commented code has removed. Issue link left